### PR TITLE
docs: parse config.cc properties as rst text

### PR DESCRIPTION
**Pull request by dgarcia360**
Related comment https://github.com/scylladb/scylladb/pull/15860#issuecomment-1792749298

This enhancement formats descriptions in `config.cc` using the standard markup language reStructuredText (RST).

By doing so, it improves the rendering of these descriptions in the documentation, allowing you to use various directives like admonitions, code blocks, ordered lists, and more.

##  Important: Test ``config.cc`` changes before merging

I couldn't manage to run the ScyllaDB test suite locally. Although I've only edited descriptions in `config.cc`,  I recommend the reviewer of this PR to test the changes done before merging.

## How to preview the changes

1. Build the docs.
2. In your browser, open /reference/configuration-parameters
3. Docs should rendered without errors.

### CODE REVIEW

**Comment by dgarcia360** [comment]
cc/ @annastuchlik @tchaikov

**Comment by denesb** [comment]
@dgarcia360 needs rebase.

File: db/config.cc
Comment (tchaikov) [inline db/config.cc:948]: thank you for fixing the unbalanced parenthesis.

File: docs/_ext/scylladb_cc_properties.py
Comment (tchaikov) [inline docs/_ext/scylladb_cc_properties.py:189]: ditto.

File: docs/_ext/scylladb_cc_properties.py
Comment (tchaikov) [inline docs/_ext/scylladb_cc_properties.py:188]: yeah, i think we could use `textwrap` or use jinja's builtin filter for indenting text.. but if it is not used, could drop it from the top of this file.

Comment (tchaikov) [reply docs/_ext/scylladb_cc_properties.py:188]: and could just return `''.join(cleaned_lines)` directly without the help of this variable.

**Review by tchaikov — APPROVED** [review]
modulo some nits, lgtm in general.

**Comment by dgarcia360** [comment]
@denesb @annastuchlik @tchaikov  rebase done and comments applied

File: db/config.cc
Comment (tchaikov) [inline db/config.cc:810]: @dgarcia360 could you please revert this change? i think "chosen" is correct. see https://www.merriam-webster.com/dictionary/choose

**Review by tchaikov — CHANGES_REQUESTED** [review]
left a comment regarding choosen versus chosen.

File: db/config.cc
Comment (dgarcia360) [reply db/config.cc]: Good catch! I'm not sure how the extra "o" got there.

**Comment by dgarcia360** [comment]
@tchaikov Fixed

**Comment by avikivity** [comment]
Please put the descriptive text in the cover letter into the commit changelog itself.

File: db/config.cc
Comment (avikivity) [inline db/config.cc:417]: Note the help text is also output when you run `scylla --help`. I guess we can adjust it later to get better output.

**Comment by tchaikov** [comment]
triggered the build.

**Comment by dgarcia360** [comment]
>  Please put the descriptive text in the cover letter into the commit changelog itself.

Commit description updated: https://github.com/scylladb/scylladb/pull/16311/commits

File: db/config.cc
Comment (dgarcia360) [reply db/config.cc:417]: Good catch, I wasn't aware of that. To ensure everything looks alright, I'll try to build ScyllaDB and run the `--help`  command to check how it looks like and make any needed adjustments before merging.

Comment (tchaikov) [reply db/config.cc:417]: i applied the change, and the output of `scylla -help` can be found at https://gist.github.com/tchaikov/f56bae6920a90e4d9f9de515689f1378 . because the size of the output, i could not paste it as a comment here.

**Comment by tchaikov** [comment]
@annastuchlik i am not a maintainer. since this change touches the `config.cc`. probably we need to wait for a maintainer to give a greenlight to this change and then queue it in the next branch.

@scylladb/scylla-maint hello maintainer, could you help review this change?  does the output of https://gist.github.com/tchaikov/f56bae6920a90e4d9f9de515689f1378 look good ?

**Review by denesb — APPROVED** [review]

**Review by tchaikov — APPROVED** [review]
lgtm

**Pull request closed without merging**
